src/main/java/seedu/address/logic/commands/DeleteAttendanceCommand.java

```diff
@@ -0,0 +1,167 @@
+        }
+
+        Student targetStudent = lastShownStudentList.get(studentIndex.getZeroBased());
+        ModuleClass targetModuleClass = lastShownModuleClassList.get(moduleClassIndex.getZeroBased());
+
+        if (lessonIndex.getZeroBased() >= targetModuleClass.getLessons().size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_LESSON_DISPLAYED_INDEX);
+        }
+
+        Lesson targetLesson = targetModuleClass.getLessons().get(lessonIndex.getZeroBased());
+
+        if (!targetLesson.getAttendanceRecordList().isWeekContained(week)) {
+            throw new CommandException(Messages.MESSAGE_INVALID_WEEK);
+        }
+
+        Lesson modifiedLesson = createModifiedLesson(targetLesson, targetStudent, week);
+        ModuleClass modifiedModuleClass = createModifiedModuleClass(targetModuleClass, lessonIndex, modifiedLesson);
+        model.setModuleClass(targetModuleClass, modifiedModuleClass);
+
+        String message = String.format(MESSAGE_DELETE_ATTENDANCE_SUCCESS, week, targetStudent.getName(), targetLesson);
+        model.commit(message);
+        return new CommandResult(message);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof DeleteAttendanceCommand // instance of handles nulls
+                && moduleClassIndex.equals(((DeleteAttendanceCommand) other).moduleClassIndex)
+                && lessonIndex.equals(((DeleteAttendanceCommand) other).lessonIndex))
+                && studentIndex.equals(((DeleteAttendanceCommand) other).studentIndex)
+                && week.equals(((DeleteAttendanceCommand) other).week);
+    }
+
+    private static ModuleClass createModifiedModuleClass(ModuleClass targetModuleClass,
+                                                         Index lessonToEditIndex, Lesson lessonToUpdate) {
+        assert targetModuleClass != null;
+        assert lessonToUpdate != null;
+
+        Name moduleClassName = targetModuleClass.getName();
+        Set<UUID> studentsIds = targetModuleClass.getStudentUuids();
+        List<Lesson> lessons = new ArrayList<>(targetModuleClass.getLessons());
+        lessons.set(lessonToEditIndex.getZeroBased(), lessonToUpdate);
+
+        return new ModuleClass(moduleClassName, studentsIds, lessons);
+    }
+
+    private static Lesson createModifiedLesson (
+            Lesson targetLesson, Student targetStudent, Week targetWeek) {
+        assert targetLesson != null;
+        assert targetStudent != null;
+        assert targetWeek != null;
+
+        Map<UUID, Attendance> record = targetLesson.getAttendanceRecordList()
+                .getAttendanceRecord(targetWeek).getAttendanceRecord();
+        Map<UUID, Attendance> updatedRecord = new HashMap<>(record);
+        // delete attendance record
+        updatedRecord.remove(targetStudent.getUuid());
```

Comment:
Should we check if the attendance we want to delete exists in the first place? 

It might be a good idea to display an error message instead of showing a success message if the attendance did not exist.

Reply:
Added guard clause to check if attendance exist in the first place, and added negative test case to check for error message throwing for this scenario